mcs/class/System/Test/System.Net.Security/SslStreamTest.cs

```diff
@@ -93,6 +171,167 @@ void AuthenticateClientAndServer (bool server, bool client)
 		}
 	}
 
+	[Category("NotWorking")]    // Works in .Net, doesn't work in mono.
+	[Test()]
+	public void SelfCompatibilityTest()
+	{
+		#if __MonoCS__
```

Comment:
As long as the code is run only in the context of mono unit tests, then as you said, it's safe to assume Mono.Security is available, and taking that one step further, it's also safe to assume ```#if __MonoCS__``` will always be true.  So the ```#else``` clauses will never be used.  I could simply chop out the ```#if``` and  ```#else``` sections (eliminating all the ```#else``` blocks), and call it a day.  However, this test is was intended to be a little bit more than a unit test.  It's a mono vs .Net compatibility test.  Unfortunately, I'm not aware of any organized location where mono vs .Net compatibility tests exist.  If there is one, it might be nice to contribute the more generalized compatibility test there, instead of just the unit test.

As discussed on the dev list in the aforementioned thread, this unit test is a trimmed down specific instance of the MonoSslStreamServerBug complete mono vs .Net SslStream compatibility test, which requires 4 machines (2 mono and 2 windows), testing all the combinations of client & server to see which ones are compatible with each other.  https://github.com/rahvee/MonoSslStreamServerBug

It was important to make a single code base able to build and run on mono, and also on windows without mono.  It just so happens that one specific test - the mono client connecting to mono server with intermediate cert - aka the mono self compatibility test - can be run on a single mono system talking to itself, so it's possible to run the mono self compatibility test as a unit test.  It also just so happens, the self compatibility test fails, and the root cause is the same as the other compatibility test failures, which makes it valuable as a unit test.

Since the mono self compatibility test has been copied from MonoSslStreamServerBug into the mono unit test code base (this pull request), it is no longer in a context where the ```#else``` clause will ever be used.  It would actually be completely fine to strip out all the ```#if``` and the ```#else``` clauses from this test.

If you like it better that way, I'll happily strip the ```#if ... #else``` clauses.  So all the code that remains is just mono code, specifically for mono unit testing and nothing else.